src/plugins/customBorders/customBorders.js

```diff
@@ -364,25 +363,169 @@ class CustomBorders extends BasePlugin {
+      if (customSelection.cellRange === null) {
+        this.hot.selection.highlight.customSelections.splice(index, 1);
+        return false; // breaks forAll
+      }
+    });
+  }
+
+  /**
+    * Clear borders.
+    *
+    * @private
+    */
+  clearBorders() {
+    arrayEach(this.savedBorders, (border) => {
+      let borderId = border.id;
+
+      this.clearBordersFromSelectionSettings(borderId);
+    });
+  }
+
+  /**
+  * Splice border from savedBorders.
+  *
+  * @private
+  * @param {String} borderId Border id name as string.
+  */
+  spliceBorder(borderId) {
+    const index = arrayMap(this.savedBorders, (border) => border.id).indexOf(borderId);
+
+    if (index > -1) {
+      this.savedBorders.splice(index, 1);
+    }
+  }
+
+  /**
+  * Check if an border already exists in the savedBorders array, and if true update border in savedBorders.
+  *
+  * @private
+  * @param {Object} border Object with `row` and `col`, `left`, `right`, `top` and `bottom`, `id` and `border` ({Object} with `color`, `width` and `cornerVisible` property) properties.
+  *
+  * @return {Boolean}
+  */
+  checkSavedBorders(border) {
+    let check = false;
+
+    arrayEach(this.savedBorders, (savedBorder, index) => {
+      if (border.id === savedBorder.id) {
+        this.savedBorders[index] = border;
+
+        check = true;
+      }
+    });
+
+    return check;
+  }
+
+  /**
+  * Check if an border already exists in the customSelections, and if true call toggleHiddenClass method.
+  *
+  * @private
+  * @param {Object} border Object with `row` and `col`, `left`, `right`, `top` and `bottom`, `id` and `border` ({Object} with `color`, `width` and `cornerVisible` property) properties.
+  * @param {String} place Coordinate where add/remove border - `top`, `bottom`, `left`, `right` and `noBorders`.
+  *
+  * @return {Boolean}
+  */
+  checkCustomSelectionsFromContextMenu(border, place, remove) {
+    let check = false;
+
+    arrayEach(this.hot.selection.highlight.customSelections, (customSelection) => {
+      if (border.id === customSelection.settings.id) {
+        objectEach(customSelection.instanceBorders, (borderObject) => {
+          borderObject.toggleHiddenClass(place, remove);
+        });
+
+        check = true;
+      }
+    });
+
+    return check;
+  }
+
+  /**
+  * Check if an border already exists in the customSelections, and if true reset cellRange.
+  *
+  * @private
+  * @param {Object} border Object with `row` and `col`, `left`, `right`, `top` and `bottom`, `id` and `border` ({Object} with `color`, `width` and `cornerVisible` property) properties.
+  * @param {CellRange} cellRange
+  *
+  * @return {Boolean}
+  */
+  checkCustomSelections(border, cellRange) {
+    let check = false;
+
+    arrayEach(this.hot.selection.highlight.customSelections, (customSelection) => {
+      if (border.id === customSelection.settings.id) {
+        customSelection.cellRange = cellRange;
+
+        check = true;
```

Comment:
Here you can add `return false` statement to stop the loop after detected the border collision.

Reply:
Done.